plugins/cni: wait for AWS to drop its CNI configuration file

squeed opened the pull request:
The script originally just exited and assumed the kubelet would restart it, but this is quite disruptive, since that restarts the agent as well.

Additionally, it was discovered that some AWS installations need the service network to be functioning before the AWS CNI file is written, leaving clusters in a broken state.

So, change the script to retry looking for the AWS CNI file itself.

Fixes: #21243

```release-note
Fixes cilium startup on certain AWS-VPC clusters.
```

sayboras commented on plugins/cilium-cni/cni-install.sh:
```diff
@@ -110,7 +110,18 @@ find "${CNI_CONF_DIR}" -maxdepth 1 -type f \
 case "$CILIUM_CNI_CHAINING_MODE" in
 "aws-cni")
-  if [ $(find "${CNI_CONF_DIR}" -maxdepth 1 -type f -name '*.conf' -or -name '*.conflist' -or -name '*.cilium_bak' | wc -l) -eq 0 ]; then
+  tries=30
```
any reason to have total time as 150 seconds ?

squeed replied:
I mean, the ideal answer is "never", but kubelet doesn't like it when postStart hooks never exit. I picked something at random; don't have a lot of evidence here either way :-(

sayboras reviewed (approved):
LGTM :heavy_check_mark:

squeed committed:
plugins/cni: wait for AWS to drop its CNI configuration file

The script originally just exited and assumed the kubelet would restart
it, but this is quite disruptive, since that restarts the agent as well.

Additionally, it was discovered that some AWS installations need the
service network to be functioning before the AWS CNI file is written,
leaving clusters in a broken state.

So, change the script to retry looking for the AWS CNI file itself.

Fixes: #21243

Signed-off-by: Casey Callendrello <cdc@isovalent.com>
